### PR TITLE
`General`: Enhance mobile appearance of various charts

### PR DESCRIPTION
**Pull request by willergeorg**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [ ] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context

Currently, many charts do not look good on smaller screens, e.g. smartphones or tablets. This is reducing the user experience for respective users.
### Description

I improved the appearance of various charts for mobile layouts (at least I tried to). The following views are enhanced by this PR:

- Course details
- Course scores
- Course statistics
- Exam statistics
- Exercise details
- Exercise statistics
- Quiz statistics

Further (smaller) improvements:

- The labels of the vertical bar charts on the course- and exercise statistics containing natural language (e.g. weekdays) are now instant translation sensitive.
- The bar labels of various vertical bar charts now rotate from horizontal to 45 deg if the screen gets smaller
- I removed an unnecessary directive declaration for the quiz statistics
- I removed the possibility to show the active students line chart in fullscreen. The tooltips are not visible in fullscreen. Thus, the fullscreen mode does not add much benefit to the user.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with at least one quiz, one other exercise and an exam with participations

1. Log in to Artemis
2. Navigate to Course Administration
3. Check the views with different screen sizes. Make sure everything is readable and nothing looks distorted.
4. For the course and exercise statistics: Check that the bar labels translate directly if you change the language settings.
5. If you have any suggestions, let me know. I cannot promise to be able to fulfill them but I can try.

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| question-statistic.component.ts | 65% | 85.22% |
| quiz-point-statistic.component.ts | 75.43% | 93.1% |
| quiz-statistic.component.ts | 72.22% | 90.9% |
| quiz-statistics.ts | 87.5% | 97.56% |
| participant-scores-distribution.component.ts | 85.48% | 93.96% |
|  statistics-graph.component.ts | 78.26% | 94.31% |

### Screenshots

<details>
<summary>Course details:</summary>

artemis-test1 artemis in tum de_course-management_7(iPhone XR)

</details>

<details>
<summary>Course scores:</summary>

localhost_9000_course-management_5_scores(iPhone 12 Pro)
</details>

<details>
<summary>Course statistics:</summary>

localhost_9000_course-management_5_course-statistics(iPhone 12 Pro)
</details>

<details>
<summary>Exam statistics:</summary>

localhost_9000_course-management_5_exams_9_scores(iPhone 12 Pro)
</details>

<details>
<summary>Exercise details:</summary>

localhost_9000_course-management_5_modeling-exercises_75(iPhone 12 Pro)
</details>

<details>
<summary>Exercise statistics:</summary>

localhost_9000_course-management_5_text-exercises_79(iPhone 12 Pro)
</details>

<details>
<summary>Quiz statistics:</summary>

localhost_9000_course-management_5_text-exercises_79(iPhone 12 Pro) (1)

localhost_9000_course-management_5_text-exercises_79(iPhone 12 Pro) (2)

localhost_9000_course-management_5_text-exercises_79(iPhone 12 Pro) (3)

localhost_9000_course-management_5_quiz-exercises_131_dnd-question-statistic_53(iPhone 12 Pro)

localhost_9000_course-management_5_quiz-exercises_159_sa-question-statistic_60(iPhone XR)

</details>

### CODE REVIEW

**Review by SimonPaul02 — COMMENTED** [review]
Found  this issue with @yuling-sun :
In the course management, there is a big white box (where the diagram was before) in most of the courses. It takes up too much space on my real mobile device. Would be good if it could be removed
https://user-images.githubusercontent.com/68793520/169690461-8ab7e43b-b02e-4b44-9c21-93101968cf39.mp4

**Review by yuling-sun — COMMENTED** [review]
The links at the bottom suddenly jump when the screen is made smaller. The link for Release Notes also suddenly disappears. Discovered this with @SimonPaul02

https://user-images.githubusercontent.com/102761130/169690972-bcc1058b-7ee6-40ba-9fc9-a78d35aae863.mp4

.

**Review by SimonPaul02 — COMMENTED** [review]
Found this issue with @yuling-sun :
The whitespace is used inefficiently at some widths, there is a huge gap, while the rest is shrinking. Also, it looks a bit right-shifted for me

https://user-images.githubusercontent.com/68793520/169691221-ae4ec8ee-731e-4a48-818d-5672902131fb.mp4

**Review by yuling-sun — COMMENTED** [review]
This statistics view also looks weird in general. Things like the Maximum points looks really out of place compared to the other elements.

**Comment by ge65cer** [comment]
> The links at the bottom suddenly jump when the screen is made smaller. The link for Release Notes also suddenly disappears.

I think that is intended as otherwise the footer would be fixed at the bottom on mobile devices, taking up too much screen space. 
Also, this PR is related to fixing the charts -- for general responsive issues refer to #5128.

**Comment by willergeorg** [comment]
> Found this issue with @yuling-sun : In the course management, there is a big white box (where the diagram was before) in most of the courses. It takes up too much space on my real mobile device. Would be good if it could be removed
> 
>  Zoom.Meeting.2022-05-22.12-10-50_Trim.2.mp4

As far as I can tell, this white box is not related to the chart. The chart simply disappears for smaller screens. I assume that the course cart has a minimum height to display information (e.g. by uncollapsing the "past exercises" part). This whitespace might be inconvenient on mobile screens, but it is not in scope of this PR as it focuses on the appearance of charts themselves. As @ge65cer  already suggested, please refer to #5128 for such issues :)

**Review by cremertim — APPROVED** [review]
tested on ts 1, lgtm

**Review by max-bergmann — DISMISSED** [review]
Tested on TS2. Works as expected

File: src/main/webapp/app/exercises/quiz/manage/statistics/drag-and-drop-question-statistic/drag-and-drop-question-statistic.component.ts [hunk 1]
```diff
@@ -18,7 +18,11 @@ import { faCheckCircle, faSync, faTimesCircle } from '@fortawesome/free-solid-sv
     selector: 'jhi-drag-and-drop-question-statistic',
     templateUrl: './drag-and-drop-question-statistic.component.html',
     providers: [QuizStatisticUtil, DragAndDropQuestionUtil],
-    styleUrls: ['../quiz-point-statistic/quiz-point-statistic.component.scss', './drag-and-drop-question-statistic.component.scss'],
+    styleUrls: [
+        '../../../../../shared/chart/vertical-bar-chart.scss',
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/exercises/quiz/manage/statistics/drag-and-drop-question-statistic/drag-and-drop-question-statistic.component.ts:22]: I hate such long relative paths, but this is not your fault :D

File: src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.html [hunk 2]
```diff
@@ -1,83 +1,80 @@
...
+            class="col-1 d-flex align-items-center switch-forward"
+            role="button"
+            (click)="switchTimeSpan(RIGHT)"
+            [ngClass]="{ hidden: !(currentSize > 10 + currentPeriod) }"
+        ></fa-icon>
+        <ng-template #placeholder>
+            <div class="switch-forward-placeholder"></div>
+        </ng-template>
+        <div class="col-lg-2 d-flex flex-column align-items-center justify-content-center">
+            <div *ngIf="typeSet.size > 0" class="filter mb-3 mt-3" aria-label="Filter Dropdown" ngbDropdown>
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.html:50]:
```suggestion
            <div *ngIf="typeSet.size > 0" class="filter my-3" aria-label="Filter Dropdown" ngbDropdown>
```

File: src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.scss [hunk 3]
```diff
@@ -40,3 +44,14 @@
 .best-color {
     background: var(--stat-av-sc-legend-best);
 }
+
+.chart-container {
+    width: 83.33%;
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/shared/statistics-graph/statistics-average-score-graph.component.scss:49]: I suggest using non-decimal percentages as I am not sure how these are handled by the browser :thinking:

File: src/main/webapp/app/shared/statistics-graph/statistics-graph.component.html [hunk 4]
```diff
@@ -1,28 +1,28 @@
...
-            [yAxisTickFormatting]="yAxisTickFormatting"
-            [animations]="false"
-        >
-            <ng-template #tooltipTemplate let-model="model">
-                <b>{{ model.name }}</b> <br />
-                <span>{{ tooltipTranslation | artemisTranslate }}: {{ model.value }}</span>
-            </ng-template>
-        </ngx-charts-bar-vertical>
+    <div class="row col-xl-11">
+        <fa-icon [icon]="faArrowLeft" size="2x" class="col-1 d-flex justify-content-end align-items-center ps-0 pe-0" role="button" (click)="switchTimeSpan(LEFT)"></fa-icon>
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/shared/statistics-graph/statistics-graph.component.html:6]:
```suggestion
        <fa-icon [icon]="faArrowLeft" size="2x" class="col-1 d-flex justify-content-end align-items-center px-0" role="button" (click)="switchTimeSpan(LEFT)"></fa-icon>
```

File: src/main/webapp/app/shared/statistics-graph/statistics-graph.component.html [hunk 5]
```diff
@@ -1,28 +1,28 @@
...
+        <div #containerRef class="chart col-10 ps-0 pe-0">
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/shared/statistics-graph/statistics-graph.component.html:7]:
```suggestion
        <div #containerRef class="chart col-10 px-0">
```

File: src/main/webapp/app/shared/statistics-graph/statistics-graph.component.html [hunk 6]
```diff
@@ -1,28 +1,28 @@
...
+                [yAxisTickFormatting]="yAxisTickFormatting"
+                [animations]="false"
+            >
+                <ng-template #tooltipTemplate let-model="model">
+                    <b>{{ model.name }}</b> <br />
+                    <span>{{ tooltipTranslation | artemisTranslate }}: {{ model.value }}</span>
+                </ng-template>
+            </ngx-charts-bar-vertical>
+        </div>
+        <fa-icon [icon]="faArrowRight" size="2x" class="col-1 d-flex justify-content-start align-items-center ps-0 pe-0" role="button" (click)="switchTimeSpan(RIGHT)"></fa-icon>
```
Comment (Dominik-Weinzierl) [inline src/main/webapp/app/shared/statistics-graph/statistics-graph.component.html:25]:
```suggestion
        <fa-icon [icon]="faArrowRight" size="2x" class="col-1 d-flex justify-content-start align-items-center px-0" role="button" (click)="switchTimeSpan(RIGHT)"></fa-icon>
```

**Review by Dominik-Weinzierl — COMMENTED** [review]
Some minor remarks - code looks good :D

**Review by Dominik-Weinzierl — DISMISSED** [review]
Thank you for implementing my suggestions - code looks good now :+1:  (i did no testing)

**Comment by willergeorg** [comment]
> The arrows of the active student chart get cut off on mobile (didn't happen before).
> 
> 

I fixed the layout in 3cec765. I also updated the corresponding screenshot in the PR description. Please check this again.

**Comment by SimonPaul02** [comment]
Tested on TS1, thank you for implementing some changes! But we found some things that don't satisfy us yet

**Comment by SimonPaul02** [comment]
The release notes now disappear in mobile view, but I guess that's a feature.

**Comment by SimonPaul02** [comment]
The Maximum Points Button could be placed more to the left. It looks weird there. 
If you place the max. point button to the left, it might still look weird because of the big gap in the centre. So you have to think about a different implementation, for example changing the button layout completely.
Found this issue with @yuling-sun 

https://user-images.githubusercontent.com/68793520/173183348-1bad2af7-b12b-4920-9cd6-1f57dae567af.mp4

**Review by yuling-sun — COMMENTED** [review]
Thanks for implementing some of our comments, but the release notes at the bottom still disappear when shrinking the window, but that might be a feature. Tested with @SimonPaul02.

**Comment by SimonPaul02** [comment]
> > The arrows of the active student chart get cut off on mobile (didn't happen before).
> > 
> 
> I fixed the layout in [3cec765](https://github.com/ls1intum/Artemis/commit/3cec7654c3520b2ad23b66dbaab0eee94eabb825). I also updated the corresponding screenshot in the PR description. Please check this again.

It doesn't load at all both at @yuling-sun and my tests. 

https://user-images.githubusercontent.com/68793520/173183664-c2f53ffe-5212-4867-9515-b57b4674e37c.mp4

**Comment by willergeorg** [comment]
> The Maximum Points Button could be placed more to the left. It looks weird there. If you place the max. point button to the left, it might still look weird because of the big gap in the centre. So you have to think about a different implementation, for example changing the button layout completely. (But maybe it won't look as weird as expected😉) Found this issue with @yuling-sun
> 
>  Courses.-.Google.Chrome.2022-06-11.11-56-35_Trim.2.mp4

Thanks for having another look. As I already tried to explain [the first time](https://github.com/ls1intum/Artemis/pull/5115#issuecomment-1138364640), this PR focuses on the appearance of **charts** in Artemis. This does not include the overall layout of the quiz statistics, nor the appearance of the page footer. I agree, that the quiz statistics layout might be improvable (for example the whitespace you already mentioned), but this is not in scope of this PR. Please be aware of this when testing the changes. :)

**Comment by willergeorg** [comment]
> > > The arrows of the active student chart get cut off on mobile (didn't happen before).
> > > 
> > 
> > 
> > I fixed the layout in [3cec765](https://github.com/ls1intum/Artemis/commit/3cec7654c3520b2ad23b66dbaab0eee94eabb825). I also updated the corresponding screenshot in the PR description. Please check this again.
> 
> It doesn't load at all both at @yuling-sun and my tests.
> 
>  Courses.-.Google.Chrome.2022-06-11.12-10-36_Trim.mp4

Thanks for reporting the issue. It is unrelated to the changes done in this PR. The corresponding logic is fixed in #5192.

**Review by max-bergmann — APPROVED** [review]
Tested on TS2 again. Looks good

**Review by ge65cer — APPROVED** [review]
Thanks for fixing the issue I mentioned, approved!

**Review by cremertim — APPROVED** [review]
tested locally, works as expected

**Review by rhubinger — APPROVED** [review]
Tested on TS2. Everything worked as expected

**Review by yuling-sun — APPROVED** [review]
Approve.

**Review by Dominik-Weinzierl — APPROVED** [review]
Code looks reasonable to me 👍 Thank you for implementing my suggestions!

**Pull request merged**